asaplib/pca/ml_kpca.py

```diff
@@ -9,6 +9,207 @@
 import scipy.linalg as salg
 
 
+class KernelPCA:
+    def __init__(self, ndim=2):
+        """KernelPCA with precomputed kernel for now only
+
+        Parameters
+        ----------
+        ndim : int, default=2
+            number of dimensions to project to
+        """
+
+        # essential
+        self.ndim = ndim
+
+        # kernel and projection related properties
+        self.vectors = None
+        self.colmean = None
+        self.mean = None
+        self.center_kmat = None
+        self._lambdas = None
+        self._alphas = None
+        self._m = None
+
+        # state properties
+        self._fitted = False
+
+    @staticmethod
+    def center_square(kernel):
+        """Centering of a kernel matrix, with additional centering info
+
+        Parameters
+        ----------
+        kernel : array-like, (shape=MxM)
+            Kernel matrix
+
+        Returns
+        -------
+        colmean : ndarray, (shape=M)
+            mean of columns from kernel matrix
+        mean : float
+            mean of the entire kernel matrix
+        centered_kernel : ndarray, (shape=MxM)
+            kernel centered in feature space
+        """
+        colmean = np.mean(kernel, axis=1)
+        mean = np.mean(colmean)
+        # centered_kernel = kernel - colmean - np.reshape(colmean, (len(colmean), 1)) + mean
+        centered_kernel = kernel - colmean - np.atleast_2d(colmean).T + mean
+        return colmean, mean, centered_kernel
+
+    def _center_test_kmat(self, ktest):
```

Comment:
What's the rationale for indicating that this method is private? (Just out of interest!)

Reply:
fair, I thought that anything that is used internally and have little point being used externally can stay private. This is called by `transform` and no need for the user to use at all

Reply:
Makes sense!